react/features/transcribing/middleware.js

```diff
@@ -0,0 +1,101 @@
+// @flow
+
+import { MiddlewareRegistry } from '../base/redux';
+
+import {
+    _TRANSCRIBER_LEFT,
+    DIAL_TRANSCRIBER,
+    STOP_TRANSCRIBING
+} from './actionTypes';
+import {
+    potentialTranscriberJoined,
+    transcriberJoined,
+    transcriberLeft,
+    dialError,
+    showPendingTranscribingNotification,
+    hidePendingTranscribingNotification,
+    showStoppedTranscribingNotification,
+    showTranscribingError
+} from './actions';
+import {
+    HIDDEN_PARTICIPANT_JOINED,
+    HIDDEN_PARTICIPANT_LEFT,
+    PARTICIPANT_UPDATED
+} from './../base/participants';
+
+declare var APP: Object;
+
+const TRANSCRIBER_DIAL_COMMAND = 'jitsi_meet_transcribe';
+const TRANSCRIBER_DISPLAY_NAME = 'Transcriber';
+
+/**
+ * Implements the middleware of the feature transcribing.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+// eslint-disable-next-line no-unused-vars
+MiddlewareRegistry.register(store => next => action => {
+    const {
+        isDialing,
+        isTranscribing,
+        transcriberJID,
+        potentialTranscriberJIDs
+    } = store.getState()['features/transcribing'];
+
+    const { conference } = store.getState()['features/base/conference'];
+
+    switch (action.type) {
+    case DIAL_TRANSCRIBER:
+        if (!(isDialing || isTranscribing)) {
+            store.dispatch(showPendingTranscribingNotification());
+
+            conference.room.dial(TRANSCRIBER_DIAL_COMMAND).catch(
+                () => {
+                    // do nothing on success
+                },
+                () => {
+                    store.dispatch(dialError());
+                    store.dispatch(hidePendingTranscribingNotification());
+                    store.dispatch(showTranscribingError());
+                }
+            );
+        }
+        break;
+    case STOP_TRANSCRIBING:
+        if (isTranscribing) {
+            APP.conference._room.kickParticipant(transcriberJID);
```

Comment:
You might also want to swap this reference to the global conference for the locally scoped conference.